cloudbaseinit/osutils/windows.py

```diff
@@ -799,6 +799,8 @@ def set_ntp_client_config(self, ntp_hosts):
                 'w32tm failed to configure NTP.\nOutput: %(out)s\nError:'
                 ' %(err)s' % {'out': out, 'err': err})
 
+    @retry_decorator.retry_decorator(
```

Comment:
I would advise to set the service to automatic delayed start in these cases. If the OS does not see the adapter, there is a problem with the nic drivers or a concurrency issue between the cloudbase-init service and the networking service.